Composer/packages/client/src/store/action/error.ts

```diff
@@ -12,9 +14,18 @@ export const setError: ActionCreator = ({ dispatch }, error) => {
   });
 };
 
-export const setMessage: ActionCreator = ({ dispatch }, message) => {
+export const setMessage: ActionCreator = debounce(({ dispatch }, message) => {
   dispatch({
     type: ActionTypes.SET_MESSAGE,
     payload: message,
   });
-};
+
+  setTimeout(
+    () =>
+      dispatch({
+        type: ActionTypes.SET_MESSAGE,
+        payload: undefined,
+      }),
+    2000
+  );
+}, 500);
```

Comment:
To type this correctly, the debounced func should be hoisted to be a module defined function and then the action should just invoke that function. 

This is because `type debounce = <T>(fn: T, opts) => T & Cancelable` which is not `ActionCreator`. Consumers of this action should not have knowledge of its implementation (that it can be cancelled for instance).

The reason the compiler likely doesn't catch this is due to a loose typing in `BoundActionCreators`.